circleci-artifacts/build.gradle

```diff
@@ -15,4 +16,19 @@ dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter'
     testImplementation 'com.netflix.nebula:nebula-test'
     testImplementation 'com.palantir.gradle.plugintesting:gradle-plugin-testing-junit'
-}
\ No newline at end of file
+}
+
+gradleTestUtils {
+    gradleVersions = ['8.14.3']
+    configurationCacheEnabled = true
+}
+
+// Publish to maven local so that tests can directly import CircleCiArtifacts
+// Need to publish all dependencies too (providers, environment-variables)
+tasks.test.dependsOn tasks.publishToMavenLocal,
```

Comment:
this causes test to be initialized.  See how we do it in the plugin testing repo:

tasks.named('test', Test) {
    systemProperty 'projectVersion', project.version

    // Added as a jar the so in nebula-tests needs to exist in maven local
    dependsOn ':plugin-testing-core:publishToMavenLocal'
    dependsOn ':gradle-plugin-testing-junit:publishToMavenLocal'
   ...
}